packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -528,6 +528,7 @@ export default class TradeStore extends BaseStore {
             purchase_spot_barrier.isSingleBarrier = true;
             purchase_spot_barrier.updateBarrierColor(this.root_store.ui.is_dark_mode_on);
             this.barriers.push(purchase_spot_barrier);
+            this.root_store.portfolio.barriers.push(purchase_spot_barrier);
```

Comment:
We need to update `barriers` of `portfolio-store` in core in order to get updated barriers on `trade.jsx` file

Reply:
@mahdiyeh-fs there should be a way to update store variable in the store itself..

Reply:
since the action of changing barriers is in the trade store and we need to update portfolio store with trade store, the best way is to update it from here.